AWSS3Tests/AWSS3TestHelper.m

```diff
@@ -0,0 +1,141 @@
+        } else {
+            success = YES;
+        }
+        return nil;
+    }] waitUntilFinished];
+    
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:10]];
+    
+    return success;
+}
+
++ (BOOL)deleteBucketWithName:(NSString *)bucketName {
+    AWSS3 *s3 = [AWSS3 defaultS3];
+    
+    AWSS3ListObjectVersionsRequest *listObjectVersionsRequest = [AWSS3ListObjectVersionsRequest new];
+    listObjectVersionsRequest.bucket = bucketName;
+    
+    [[[s3 listObjectVersions:listObjectVersionsRequest] continueWithSuccessBlock:^id(AWSTask *task) {
+        NSMutableArray *tasks = [NSMutableArray new];
+        
+        AWSS3ListObjectVersionsOutput *listObjectVersionsOutput = task.result;
+        for (AWSS3ObjectVersion *version in listObjectVersionsOutput.versions) {
+            AWSS3DeleteObjectRequest *deleteObjectRequest = [AWSS3DeleteObjectRequest new];
+            deleteObjectRequest.bucket = bucketName;
+            deleteObjectRequest.key = version.key;
+            deleteObjectRequest.versionId = version.versionId;
+            AWSTask *task = [s3 deleteObject:deleteObjectRequest];
+            [tasks addObject:task];
+        }
+        for (AWSS3DeleteMarkerEntry *deleteMarker in listObjectVersionsOutput.deleteMarkers) {
+            AWSS3DeleteObjectRequest *deleteObjectRequest = [AWSS3DeleteObjectRequest new];
+            deleteObjectRequest.bucket = bucketName;
+            deleteObjectRequest.key = deleteMarker.key;
+            deleteObjectRequest.versionId = deleteMarker.versionId;
+            AWSTask *task = [s3 deleteObject:deleteObjectRequest];
+            [tasks addObject:task];
+        }
+        
+        return [AWSTask taskForCompletionOfAllTasks:tasks];
+    }] waitUntilFinished];
+    
+    AWSS3DeleteBucketRequest *deleteBucketReq = [AWSS3DeleteBucketRequest new];
+    deleteBucketReq.bucket = bucketName;
+    
+    __block BOOL success = NO;
+    [[[s3 deleteBucket:deleteBucketReq] continueWithBlock:^id(AWSTask *task) {
+        if (task.error) {
+            success = NO;
+        } else {
+            success = YES;
+        }
+        return nil;
+    }] waitUntilFinished];
+    
+    return success;
+}
+
++ (void)deleteAllObjectsFromBucket:(NSString *)bucketName {
+    AWSS3 *s3 = [AWSS3 defaultS3];
+    
+    AWSS3ListObjectsRequest *listObjectsRequest = [AWSS3ListObjectsRequest new];
+    listObjectsRequest.bucket = bucketName;
+    
+    [[[s3 listObjects:listObjectsRequest] continueWithBlock:^id(AWSTask *task) {
+        AWSS3ListObjectsOutput *output = task.result;
+        
+        for (AWSS3Object *object in output.contents) {
+            // Delete the object
+            AWSS3DeleteObjectRequest *deleteObjectRequest = [AWSS3DeleteObjectRequest new];
+            deleteObjectRequest.bucket = bucketName;
+            deleteObjectRequest.key = object.key;
+            
+            [[s3 deleteObject:deleteObjectRequest] continueWithBlock:^id(AWSTask *task) {
```

Comment:
Consider a batch delete here instead of individual single deletes: https://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html

Reply:
Changed to multi delete.